kernel/fs/fs_ramdisk.c

```diff
@@ -103,11 +103,11 @@ static mutex_t rd_mutex;
 
 /* Search a directory for the named file; return the struct if
    we find it. Assumes we hold rd_mutex. */
-static rd_file_t * ramdisk_find(rd_dir_t * parent, const char * name, int namelen) {
+static rd_file_t *ramdisk_find(rd_dir_t *parent, const char *name, size_t namelen) {
     rd_file_t   *f;
 
     LIST_FOREACH(f, parent, dirlist) {
-        if(!strncasecmp(name, f->name, namelen))
+        if(!strncasecmp(name, f->name, namelen) && strlen(f->name) == namelen)
```

Comment:
Good catch. Two minor suggestions. 1) toss parens around the strlen comparison so that it's clearer. 2) swap it to the front since it's a faster check to do.
Something like:
`if((strlen(f->name) == namelen) && !strncasecmp(name, f->name, namelen))`

Reply:
Done